src/components/learner-credit-management/data/utils.js

```diff
@@ -551,3 +555,97 @@ export const isLmsBudget = (
   activeIntegrationsLength,
   isUniversalGroup,
 ) => activeIntegrationsLength > 0 && isUniversalGroup;
+
+/**
+ * Determines if the course has already started. Mostly used around text formatting for tense
+ *
+ * @param start
+ * @returns {boolean}
+ */
+export const hasCourseStarted = (start) => dayjs(start).isBefore(dayjs());
+
+/**
+ * Returns assignable course runs within the threshold of within the subsidies expiration date
+ * offset by the DAYS_UNTIL_ASSIGNMENT_ALLOCATION_EXPIRATION constant. It sorts it from the soonest expiring
+ * enroll-by date and the enroll-by date and upgrade deadline has been normalized to ISO format.
+ *
+ * @param courseRuns
+ * @param subsidyExpirationDatetime
+ * @param isLateRedemptionAllowed
+ * @returns {*}
+ */
+export const getAssignableCourseRuns = ({ courseRuns, subsidyExpirationDatetime, isLateRedemptionAllowed }) => {
+  const clonedCourseRuns = courseRuns.map(courseRun => ({
+    ...courseRun,
+    enrollBy: dayjs.unix(courseRun.enrollBy).toISOString(),
+    upgradeDeadline: dayjs.unix(courseRun.upgradeDeadline).toISOString(),
+  }));
+  const assignableCourseRunsFilter = ({ enrollBy, isActive }) => {
+    const enrollByDateThreshold = dayjs(enrollBy).isBefore(
+      Math.max(
+        dayjs(subsidyExpirationDatetime).subtract(MAX_ALLOWABLE_REFUND_THRESHOLD_DAYS, 'days').toDate(),
+        dayjs().add(DAYS_UNTIL_ASSIGNMENT_ALLOCATION_EXPIRATION, 'days').toDate(),
+      ),
+    );
```

Comment:
Logic changes to conform to my assumed goals above:
```suggestion
    const enrollByDateThreshold = dayjs(enrollBy).isBefore(
      dayjs(subsidyExpirationDatetime).subtract(MAX_ALLOWABLE_REFUND_THRESHOLD_DAYS, 'days').toDate()
    ) && dayjs(enrollBy).isAfter(
      dayjs().add(DAYS_UNTIL_ASSIGNMENT_ALLOCATION_EXPIRATION, 'days').toDate(),
    );
```

Reply:
Great callout. By looking at the min value as opposed to the max value, we are taking into account the suggested changes 👍🏽